fms/models/__init__.py

```diff
@@ -0,0 +1,39 @@
+import os
+from pathlib import Path
+import re
+from typing import Callable, List
+import torch
+from torch import nn
+
+__models = {}
+
+
+def register_model(architecture, variant, factory):
+    variants = {}
+    if architecture in __models:
+        variants = __models[architecture]
+    variants[variant] = factory
+    __models[architecture] = variants
+
+
+def list_models():
+    return list(__models.keys())
+
+
+def list_variants(architecture: str):
+    return list(__models[architecture].keys())
+
+
+def get_model(architecture: str, variant: str, extra_args: dict = {}) -> nn.Module:
```

Comment:
We may want to add a get_tokenizer for this as well

Reply:
I think I prefer keeping that separate. we have a distinct helper function to get a tokenizer. I don't want to assume that the tokenizer is a file in the model directory (and this function doesn't actually refer to a model directory).

the tokenizer also seems more related to weight loading than model instantiation (e.g. when training from scratch someone might want to train their own tokenizer).